Move one level up the sections in the configuration file

monicasarbu opened the pull request:
Based on the changed done in #1490, I am proposing the following structure of the configuration file where I move all the sections one level up all. 

Implements part of https://github.com/elastic/beats/issues/1417.

The changes should affect the following configuration files:

- [x] libbeat.yml
- [x] packetbeat.yml
- [x] topbeat.yml
- [x] filebeat.yml
- [x] winlogbeat.yml

ruflin commented on packetbeat/packetbeat.yml:
```diff
@@ -24,132 +24,131 @@ packetbeat.flows:
-    # Configure the ports where to listen for NFS traffic. You can disable
-    # the NFS protocol by commenting out the list of ports.
-    ports: [2049]
+packetbeat.protocols.icmp:
```
Isn't the "move one level up" already done with packetbeat.protocols?

monicasarbu replied:
I moved them one more level up. I think it's easier to read the configuration file like this. I think it's similar with what ES has now.

ruflin commented:
@monicasarbu I think that is in general a good change. What I'm not sure about is our "logic"? When do we exactly use the dot nation, and when exactly sub points? Would be good to have a common logic that we can always apply.

monicasarbu commented:
@ruflin @tsg @urso @andrewkroh I finished changing all the config files.  Please let me know what you think about the new structure.

urso commented on topbeat/etc/beat.yml:
```diff
@@ -10,17 +11,17 @@ topbeat:
-    cpu_per_core: false
+  
+  # per system statistics, by default is true
+  stats.system: true
```
for topbeat I think it makes sense to have a fully flat config:

```
topbeat.period: ...

topbeat.procs: 

topbeat.stats.system: ...
```

ruflin replied:
Or the follwoing would als be fine from my perpsective:
```
topbeat.period
topbeat.procs
topbeat.stats:
  system: ...
  process: ...
```

urso commented:
I like the new structure, but change looks very minimal. I don't think this prevents people from messing up filebeat prospector config.

packetbeat, filebeat look good. For topbeat/winlogbeat i'd rather flatten more.

ruflin commented on filebeat/etc/beat.yml:
```diff
@@ -1,176 +1,175 @@
+
+######################### Filebeat general configuration #############################
+
+filebeat:
```
I quite like the idea of separating general config and prospectors. Should we even comment out "filebeat" here as all sub config options are commented out?

monicasarbu replied:
yes, good spot

ruflin commented on libbeat/etc/libbeat.yml:
```diff
@@ -249,19 +244,18 @@ logging:
   #to_files: false
```
Alternatively we could do:

```
logging.files:
  name:
  rotateevery....
  ...

The problem here is that I think to_syslog and to_files are somehow not fitting in our concept. Means if files is commented out it should be enabled or having an enabled flag inside would help.

ruflin commented on winlogbeat/winlogbeat.yml:
```diff
@@ -1,5 +1,7 @@
+
+##################### Winlogbeat general configuration ######################
+
 winlogbeat:
```
Shouldn't we go with `winlogbeat.event_logs` here?

ruflin commented:
Changes look good to me. See comments above. In a second step we should verify that all config options which actually exist are also in the config file. So in a third step we can potentially create a minimized option.

ruflin commented:
BTW: One idea to create the minimized option of the config: Remove all fields which are commented out in the config could work.

ruflin commented:
@monicasarbu It would be good to also update the config files we use for system tests with the above changes to make sure all the changes work as expected.

monicasarbu commented:
@ruflin @tsg @urso @andrewkroh I have updated the config files with your comments.  

@ruflin I agree with you that we need to update the system tests and also create a minimized version as we discussed. I have updated the meta ticket with these two new requests: https://github.com/elastic/beats/issues/1417

ruflin commented:
@monicasarbu One of the configtest system tests seems to fail ...
